### PR TITLE
ci: enable `flake8`

### PR DESCRIPTION
**Pull request by JoepVanlier**
**Why this PR?**
To help us not shoot ourselves in the feet.

`flake8` can hopefully help us stick to `PEP` and flag things that may become issues.

For now, I have excluded the test folders analogously to what we do with `black`. We should definitely still add them, but there were already many fixups and I didn't want this PR to be blocked on just this (since it will involve cleaning up the tests as well and that would overlap with what Tobias is doing now).

I fixed up all of the things it reported. It's many commits, but all of them are pretty simple individually. I'd recommend going commit by commit, since I clustered them by problem. Once approved, I will squash each class of error as one commit.

In the process I noticed that some dwelltime plots were untested, so I added a few tests whether those open without exception at least.

Finally, I debated whether to replace empty dictionaries with `None` or `MappingProxyType({})` and decided for the former as it would be less confusing to end users when looking at the function signature. Even though I'm aware that the latter also ensures that we don't modify the input variable.

Please also check the rules I excluded in `.flake8`. I added comments as to what and why. We can discuss whether we want to add more exclusions in the future. For now I stuck to making sure it plays nicely with `black` and to not flag some patterns that we stubbornly use (such as `*` imports).

### CODE REVIEW

File: lumicks/pylake/scan.py [hunk 1]
```diff
@@ -350,7 +350,9 @@ def plot_channel(frame):
         def post_update(image_handle, image):
             return adjustment._update_limits(image_handle, image, channel)
 
-        title_factory = lambda frame: make_image_title(self, frame, show_name=False)
+        def title_factory(frame):
+            return make_image_title(self, frame, show_name=False)
+
```
Comment (tobiasjj) [inline lumicks/pylake/scan.py:355]: 👍

**Review by tobiasjj — APPROVED** [review]
Many nice little improvements! Only few nitpicks. LGTM 🚴

File: .github/workflows/pylake_test.yml [hunk 2]
```diff
@@ -38,3 +38,7 @@ jobs:
       run: |
         pip install black==22.3.0
         black --check --diff --color .
+    - name: flake8
+      run: |
+        pip install flake8 flake8-bugbear
+        flake8 --color always .
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply .github/workflows/pylake_test.yml:44]: Doesn't harm to enforce it. I'll do it.

File: lumicks/pylake/fitting/fit.py [hunk 3]
```diff
@@ -350,9 +350,9 @@ def trial(parameters=[]):
 
         return profile
 
-    def _calculate_residual(self, parameter_values=[]):
+    def _calculate_residual(self, parameter_values=None):
         self._rebuild()
-        if len(parameter_values) == 0:
+        if parameter_values is None:
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/fitting/fit.py:355]: Doesn't play well with numpy.

You'd get `The truth value of an array with more than one element is ambiguous`.

File: lumicks/pylake/fitting/fit.py [hunk 4]
```diff
@@ -367,9 +367,9 @@ def _calculate_residual(self, parameter_values=[]):
 
         return residual
 
-    def _calculate_jacobian(self, parameter_values=[]):
+    def _calculate_jacobian(self, parameter_values=None):
         self._rebuild()
-        if len(parameter_values) == 0:
+        if parameter_values is None:
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/fitting/fit.py:372]: See above.

File: lumicks/pylake/correlated_stack.py [hunk 5]
```diff
@@ -464,22 +464,19 @@ def plot_correlated(
...
-            frame_grabber,
-            title_factory,
-            frame,
-            reduce,
+            channel_slice=channel_slice,
+            frame_timestamps=self.frame_timestamp_ranges(),
+            get_plot_data=frame_grabber,
+            title_factory=lambda frame: make_image_title(self, frame, show_name=False),
+            frame=frame,
+            reduce=reduce,
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/correlated_stack.py:479]: We should probably make a habit out of that. Especially when lambdas are involved 😅

File: lumicks/pylake/fitting/detail/utilities.py [hunk 6]
```diff
@@ -4,11 +4,12 @@
...
 def unique_idx(input_list):
     """
-    Determine unique elements of a list and return indices which reconstruct the original list from the unique elements.
+    Determine unique elements of a list and return indices which reconstruct the original list from
+    the unique elements.
     """
     unique_list = []
     [unique_list.append(x) for x in input_list if x not in unique_list]
-    inverse_list = [unique_list.index(l) for l in input_list]
+    inverse_list = [unique_list.index(x) for x in input_list]
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/fitting/detail/utilities.py:12]: Ah yeah, might as well fix that up while we are there.

File: .github/workflows/pylake_test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply .github/workflows/pylake_test.yml]: Hmm, actually, `6.0.0` doesn't support `3.7`. We should drop `3.7` soon anyway, but maybe for now we can just leave the color argument out and not add the version req.

File: lumicks/pylake/fitting/detail/utilities.py [hunk 7]
```diff
@@ -4,12 +4,13 @@
...
-    Determine unique elements of a list and return indices which reconstruct the original list from the unique elements.
+    Determine unique elements of a list and return indices which reconstruct the original list from
+    the unique elements.
     """
     unique_list = []
     [unique_list.append(x) for x in input_list if x not in unique_list]
-    inverse_list = [unique_list.index(l) for l in input_list]
-    return unique_list, inverse_list
+    inverse_indices = [unique_list.index(x) for x in input_list]
+    return unique_list, inverse_indices
```
Comment (tobiasjj) [inline lumicks/pylake/fitting/detail/utilities.py:13]: 👍

**Pull request merged**